lib/lightstep/span.rb

```diff
@@ -40,8 +42,14 @@ def initialize(
       @tracer = tracer
       self.operation_name = operation_name.to_s
       self.start_micros = start_micros
+
+      trace_id = (SpanContext === child_of ? child_of.trace_id : LightStep.guid)
       @span_context = SpanContext.new(id: LightStep.guid, trace_id: trace_id)
-      set_tag(:parent_span_guid, child_of_id) if !child_of_id.nil?
+
+      if SpanContext === child_of
```

Comment:
#nit If you move this to be before `trace_id = (SpanContext === child_of`, you can save a `SpanContext === child_of` check and the code will be a bit easier to reason with IMO